### PR TITLE
SNOW-749141: Fix uploading a file > 200M when the same filename already exists in a stage

### PR DESCRIPTION
**Pull request by sfc-gh-yixie**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #SNOW-749141. 

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.
Call get_digest even when the file already exists on a stage. The current code skips calling get_digest so the digest is None.

### CODE REVIEW

File: src/snowflake/connector/storage_client.py [hunk 1]
```diff
@@ -185,7 +185,7 @@ def get_file_header(self, filename: str) -> FileHeader | None:
     def preprocess(self) -> None:
         meta = self.meta
         logger.debug(f"Preprocessing {meta.src_file_name}")
-
+        self.get_digest()
```
Comment (sfc-gh-aling) [inline src/snowflake/connector/storage_client.py:188]: can we test it? I remember we discussed offline that we can change the chunk size to make it easier to test

File: src/snowflake/connector/storage_client.py [hunk 2]
```diff
@@ -185,7 +185,7 @@ def get_file_header(self, filename: str) -> FileHeader | None:
...
         if not meta.overwrite:
             self.get_file_header(meta.dst_file_name)  # Check if file exists on remote
```
Comment (sfc-gh-aling) [inline src/snowflake/connector/storage_client.py:190]: while this `self.get_file_header` raise error? if it raises error, then is it still correct to call `self.get_digest` ahead

Comment (sfc-gh-yixie) [reply src/snowflake/connector/storage_client.py:190]: As discussed offline, update the PR to add a new call to `get_digest` under `if not meta.overwrite:` and keep the previous `get_digest`.

File: src/snowflake/connector/storage_client.py (hunk 1 above)
Comment (sfc-gh-yixie) [reply src/snowflake/connector/storage_client.py]: Updated `test_large_put.py` to cover the this fix. 
I manually tested by using the updated test and removing the fix. The problem was reproduced. After adding back the fix with this PR, the problem disappeared.
I also manually tested with a file 250M on aws, azure and gcp.

**Review by sfc-gh-aling — APPROVED** [review]

**Comment by sfc-gh-yixie** [comment]
@sfc-gh-mkeller Not sure why code coverage is down by -0.05%. The two impacted indirect files, one has code coverage decrease, shouldn't be related to this PR.
https://app.codecov.io/gh/snowflakedb/snowflake-connector-python/pull/1470/indirect-changes

**Pull request merged**